src/hooks/useOverflowDetection/useOverflowDetection.ts

```diff
@@ -0,0 +1,26 @@
+import { useCallback, useRef } from "react";
+
+export const useOverflowDetection = <T extends HTMLElement>() => {
+  const elementsRef = useRef<T[]>([]);
+
+  const setRef = useCallback(
```

Comment:
suggestion: I got confused that `setRef` is a function that you can pass to `ref` in React, so maybe we should use some other name like, e.g. `getRefForIndex`?

```suggestion
  const getRefForIndex = useCallback(
```